lib/Random/RandomGenerator.cpp

```diff
@@ -201,10 +204,10 @@ int32_t RandomDevice::other(int32_t left, uint32_t range) {
 
   r %= range;
 
-  int32_t result = left + static_cast<int32_t>(r);
-  TRI_ASSERT(result >= left);
+  int64_t result = static_cast<int64_t>(left) + static_cast<int64_t>(r);
+  TRI_ASSERT(result >= static_cast<int64_t>(left));
   TRI_ASSERT(result < left + static_cast<int64_t>(range));
-  return result;
+  return static_cast<int32_t>(result);
```

Comment:
Can we additionally check here (via assertion) that result will fit into an `int32_t`?